### PR TITLE
Feature/speedup parquet export

### PR DESCRIPTION
**Pull request by jcharkow**
Re-implementation of PR https://github.com/PyProphet/pyprophet/pull/111

This uses duckdb for parquet export instead of pandas. This allows for memory and SQL queries to be handled by duckdb which is faster and uses less memory.

### CODE REVIEW

**Comment by jcharkow** [comment]
Currently, this method only supports a combined output file (no split based on runs) and no IPF

**Comment by jcharkow** [comment]
@singjc can you review this when you get a chance?

**Comment by grosenberger** [comment]
Looks great, thank you! Please feel free to merge when you think that it is ready.

File: pyprophet/export_parquet.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    columns['FEATURE_MS1'] = ['APEX_INTENSITY', 'VAR_MASSDEV_SCORE', 'VAR_MI_SCORE', 'VAR_MI_CONTRAST_SCORE', 'VAR_MI_COMBINED_SCORE', 'VAR_ISOTOPE_CORRELATION_SCORE', 'VAR_ISOTOPE_OVERLAP_SCORE', 'VAR_IM_MS1_DELTA_SCORE', 'VAR_XCORR_COELUTION', 'VAR_XCORR_COELUTION_CONTRAST', 'VAR_XCORR_COELUTION_COMBINED', 'VAR_XCORR_SHAPE', 'VAR_XCORR_SHAPE_CONTRAST', 'VAR_XCORR_SHAPE_COMBINED']
```
Comment (singjc) [inline pyprophet/export_parquet.py:126]: Is it worthwhile to be able to automatically retrieve some of these column names (at least for the `VAR_` scores), for the event that futures scores are added/removed?

File: pyprophet/export_parquet.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+    ## mappings
+    columns['PRECURSOR_PEPTIDE_MAPPING'] = ['PEPTIDE_ID', 'PRECURSOR_ID']
+    columns['PEPTIDE_PROTEIN_MAPPING'] = ['PROTEIN_ID']
+    if check_sqlite_table(con, "GENE") and pd.read_sql('SELECT GENE_NAME FROM GENE', con).GENE_NAME[0]!='NA':
+        columns['PEPTIDE_GENE_MAPPING'] = ['GENE_ID']
+
+    # transition level
+    if transitionLevel:
+        columns['FEATURE_TRANSITION'] = ['AREA_INTENSITY', 'TOTAL_AREA_INTENSITY', 'APEX_INTENSITY', 'TOTAL_MI', 'VAR_INTENSITY_SCORE', 'VAR_INTENSITY_RATIO_SCORE', 'VAR_LOG_INTENSITY', 'VAR_XCORR_COELUTION', 'VAR_XCORR_SHAPE', 'VAR_LOG_SN_SCORE', 'VAR_MASSDEV_SCORE', 'VAR_MI_SCORE', 'VAR_MI_RATIO_SCORE', 'VAR_ISOTOPE_CORRELATION_SCORE', 'VAR_ISOTOPE_OVERLAP_SCORE']
```
Comment (singjc) [inline pyprophet/export_parquet.py:163]: Same comment as above. I know it's currently not implemented for IPF, but I have added more IM scores to the `FEATURE_TRANSITION` table, so would be nice, not having to think about needing to manually keep track and add these scores.

File: pyprophet/main.py [hunk 3]
```diff
@@ -350,6 +350,30 @@ def export(infile, outfile, format, outcsv, transition_quantification, max_trans
...
 
+# Export to Paruqet
+@cli.command()
+@click.option('--in', 'infile', required=True, type=click.Path(exists=True), help='PyProphet input file.')
+@click.option('--out', 'outfile', required=False, type=click.Path(exists=False), help='Output parquet file.')
+@click.option('--transitionLevel', 'transitionLevel', is_flag=True, help='Whether to export transition level data as well')
+@click.option('--separate_runs/--no-separate_runs', default=True, show_default=True, help='If the input file is a merged file, create separate parquet files per run')
+@click.option('--onlyFeatures', 'onlyFeatures', is_flag=True, help='Only include precursors that have a corresponding feature')
+@click.option('--chunksize', default=1000, show_default=True, type=int, help='Amount of precursor id batch sizes to process when querying OSW file. This helps reduce memory consumption during large sql calls.')
+@click.option('--threads', default=1, show_default=True, type=int, help='Number of threads used for exporting. -1 means all available CPUs.', callback=transform_threads)
```
Comment (singjc) [inline pyprophet/main.py:361]: I think some of these args are no longer used/needed?

**Review by singjc — CHANGES_REQUESTED** [review]
Looks great, thanks! Just had some minor comments regarding the `FEATURE_{table}` score names, if some of these could be retrieved dynamically for cases when new scores are added or old ones removed.

There are some click args that are no longer used/needed. Should probably remove those.

**Comment by jcharkow** [comment]
@singjc suggestions should be addressed now

**Review by singjc — APPROVED** [review]
Thanks for the changes. Will merge now.

**Pull request merged**